airbyte-workers/src/main/java/io/airbyte/workers/protocols/airbyte/NamespacingMapper.java

```diff
@@ -40,30 +41,38 @@
  */
 public class NamespacingMapper implements Mapper<AirbyteMessage> {
 
+  private final NamespaceDefinitionType namespaceDefinition;
   private final String prefix;
 
-  public NamespacingMapper(String prefix) {
+  public NamespacingMapper(NamespaceDefinitionType namespaceDefinition, String prefix) {
+    this.namespaceDefinition = namespaceDefinition;
     this.prefix = prefix;
   }
 
   @Override
   public ConfiguredAirbyteCatalog mapCatalog(final ConfiguredAirbyteCatalog inputCatalog) {
     final ConfiguredAirbyteCatalog catalog = Jsons.clone(inputCatalog);
-    catalog.getStreams().forEach(s -> mutateStream(s.getStream(), prefix));
+    catalog.getStreams().forEach(s -> mutateStream(s.getStream(), namespaceDefinition, prefix));
     return catalog;
   }
 
   @Override
   public AirbyteMessage mapMessage(final AirbyteMessage inputMessage) {
     if (inputMessage.getType() == Type.RECORD) {
       final AirbyteMessage message = Jsons.clone(inputMessage);
+      if (namespaceDefinition != null && namespaceDefinition.equals(NamespaceDefinitionType.DESTINATION)) {
```

Comment:
spoke offline: same comment here about default behaviour possibly linking to the api spec file.